utils/stacktrace/frame.js

```diff
@@ -24,11 +24,28 @@ class Frame {
    */
   constructor(name, source, line, column) {
     this.name = name;
-    this.source = source;
+    this.source = this.dedupeSource(source);
     this.line = parseInt(line, 10);
     this.column = parseInt(column, 10);
   }
 
+  /**
+   * Dedupes source paths.
+   * Temporary fix until #27665 fixes the sourcemaps.
+   * @param {string} source
+   * @return {string}
+   */
+  dedupeSource(source) {
+    if (!source) {
+      return source;
+    }
+
+    const matches = source.match(/ampproject\/amphtml\/\d+\/(.+)\/\1\/[^\/]+$/);
```

Comment:
We can combine both the match and replace, which'll make it faster:

```js
source = source.replace(/(ampproject\/amphtml\/\d+)(\/.+)\2(\/[^\/]+)$/, '$1$2$3')
```

Reply:
Done.